---
title: "Link drug combi data to molecular background"
author: "Britta Velten"
date: "21 Oct 2016"
output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 3
---
#Introduction
Takes input from DrugCombi_Visualisation.Rmd and link molecular profiles to patients.

```{r, echo=F}
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```

```{r, echo=F}
library(ggplot2)
library(RColorBrewer)
library(Biobase)
library(abind)
library(grid)
library(gtable)
library(gridExtra)
require(pracma)
library(pheatmap)
library(reshape2)
library(dplyr)
library(lattice)
library(plyr)
library(glmnet)
library(ipflasso)
library(beeswarm)
library(SummarizedExperiment)
```

```{r, echo=F}
today<-"090817"
setwd("~/Documents/cll/MarinaDrugComb/")
datadir<-"~/Documents/cll/MarinaDrugComb/rawData"
outputdir = paste0("output", today)
if(!file.exists(outputdir)) dir.create(outputdir)
```

#Overview over external plots produced in this file:

|plot name | description |
| ------------- | ------------- |
| â€œPatientOverview.pdf" | matrix giving an overview over the patients including infromation about sex, age, somatic mutations, IGHV status, which screens they are in and their color in all other plots |
| "DiffPerConc_IGHVann.pdf" | same plot as "DiffPerConc.pdf" with asteriks markin M-CLL patients |
| "ABC_IGHVann.pdf" | same plot as "ABC.pdf" with asteriks marking M-CLL patients |
| "Curves_IGHVann.pdf" | same plot as "Curves.pdf" but without the curves for every patient instead one curve displaying median of all M-CLL patient (red), one curve displaying median of all M-CLL patient (green), one curve displaying the overall median (black). Again solid lines correspond to the measured viability value with both drugs,  dashed lines to the values from a theoretical additive model |
| "volcanoeAssoc.pdf" | Volcanoe Plot showing the p-values for association from viability values of drug combinations to the genetic features given by TP53, Trisomy12 and IGHV. |
| "HeatmapDrugxPatients.pdf" | For each combination substance used heatmap based on correlation of viabilities for the drug combintation, in which the viability values of the combined drug therapy are displayed. The patients are annotated by certain genetic features. White values in the matrix have a normalized viability value of more than upper value in legend. |
| "HeatmapDrugxPatients_zscore.pdf" | Heatmap based on centred and scaled columns |

#Data Import:  from DrugCombi_Visualtion.Rmd and molecular data about patients

```{r}
load("DrugCombi_VisualisationData.RData")
```

Loaded data from DrugCombi_VisualisationData.RData:

- 'DrugBase': contains metadata about basis drugs used (e.g. name concentrations)
- 'DrugAll': contains metadata about all drugs (e.g. name, target, id, synonyms etc)
- 'CompleteDF': contains data from all plates, wells and screens, raw values and values normalized by median of DM control wells
- 'SetUp12', 'SetUpRest' contains info about plate structure for the screens
- 'df4ana' contains values for all drug-drug pairs (effectBS) with matched one-only effect from corresponding +DMSO well (effectB) or median of DM+ wells (effectC)
-'df4anaAvreplicates' removed replicates from df4ana and replaces their values by average of the two replicates
-'patcol' color for patients

#Remove Outliers from futher analysis?
```{r}
# df4anaAvreplicates<-df4anaAvreplicatesRmOutliers
# outputdir<-paste(outputdir, "RemOutl", sep="_")
# if(!file.exists(outputdir)) dir.create(outputdir)
```



#Link to molecular data

```{r}
source("../SMART_Paper_1/addons.R")
source("../SMART_Paper_1/helperFunctions.R")

# allDataSets <- c(
#   patmeta  = "patmeta_150609.RData",    ## - patient (clinical) data
#   mut      = "mutCOM_150907.RData",     ## 454, FISH, CNV, WES all combined  
#   rna      = "e_150311.RData",          ## RNA-Seq
#   meth     = "meth_150724.RData",        ## 450k array
#   surv = "surv_160201.RData",            ## survival data
#   drugs="drugs_151007.RData"
# )

# use new version of datasets
allDataSets <- c(
  patmeta  = "patmeta_150609.RData",    ## - patient (clinical) data
  mut      = "mutCOM_150907.RData",     ## 454, FISH, CNV, WES all combined  
  rna      = "e_160829.RData",          ## RNA-Seq
  meth     = "meth_160426.RData",        ## 450k array
  surv = "surv_160201.RData",            ## survival data
  drugs="drugs_160418.RData"
)

loadRData(allDataSets, dir="/Users/bvelten/Documents/cll/var")
# e <- updateObject(e)
# meth <- updateObject(meth)
```

##Patients and molecular data available
```{r}
patientsID <- as.character(unique(df4ana$PatientID))
length(patientsID)

addMissingPatients<-function(patientID, molData){
  missing<-patientsID[which(!patientsID%in%rownames(molData))]
  print(paste("Missing patients:", ifelse(length(missing)>0,paste(missing, collapse = ", "),"none")))
  add<-matrix(NA,length(missing),ncol(molData))
  rownames(add)<-missing
  molData<-rbind(molData, add)
  molData<-molData[patientsID,]
  stopifnot(rownames(molData)==patientsID)
  return(molData)
}
```

####Methylation Data
```{r}
methData<-t(assay(meth))
rownames(methData)<-colData(meth)$PatID
methData<-addMissingPatients(patientID, methData)
rownames(methData)[apply(methData,1, function(r) any(is.na(r)))]
rownames(methData)[apply(methData,1,function(r) all(is.na(r)))]
```

####RNAseq Data
```{r}
eData <- t(assay(e))
rownames(eData)<-colData(e)$PatID
eData<-addMissingPatients(patientID, eData)
rownames(eData)[apply(eData,1, function(r) any(is.na(r)))]
checkNA((eData)[,1:100])
```

####Somatic mutations
$$
%\begin{itemize}
%\item[FP] - FISH partial - the FISH was performed on selection of cytobands
%\item[FC] - FISH complete - the FISH was performed on complete set of cytobands
%\item[4P] - 454 sequencing partial - 454seq was performed for the selection of genes
%\item[4C] - 454 sequencing complete - 454seq was performed for the complete set of genes
%\item[W] - whole exome sequencing
%\item[Sc] - SNP array of Cito type was performed
%\item[So] - SNP array of Omni type was performed
%\end{itemize}
$$
```{r, fig.width=20}
#check what is available
fData(mutCOM)
AvGen<-addMissingPatients(patientID, fData(mutCOM))

AvPat<-addMissingPatients(patientID, patmeta)
AvPat[,16:31]
#remove features with no occurences
mutCOMTh0<-channel(mutCOM, "cloneTh0")
mutCOMTh0<-mutCOMTh0[, pData(mutCOM)$Include]
genData<-exprs(mutCOMTh0)
idx <- which(colnames(genData) %in% c("del13q14_bi", "del13q14_mono"))
genData <- genData[,-idx]
colnames(genData)[which(colnames(genData)=="del13q14_any")] = "del13q14"
genData<-addMissingPatients(patientID, genData)
checkNA(genData)
minOcc <- 1
genData<-genData[,colSums(genData, na.rm=T)>=minOcc]
checkNA(genData)
```

####IGHV status
```{r}
translation <- c(`U` = 0, `M` = 1)
stopifnot(all(patmeta$`IGHV Uppsala U/M` %in% c("U","M", NA)))
IGHVData <- matrix(translation[patmeta$`IGHV Uppsala U/M`], 
                   dimnames = list(rownames(patmeta), "IGHV"), ncol = 1)
IGHVData<-addMissingPatients(patientID, IGHVData)
IGHVData<-as.matrix(IGHVData)
colnames(IGHVData)<-"IGHV"
checkNA(IGHVData)

```

####clinical data (age and sex)
```{r}
patmeta<-subset(patmeta, Diagnosis=="CLL")
gender <- ifelse(patmeta[,"Gender"]=="m",0,1)
dob<-patmeta[,"Date of birth"]
IC50date<-patmeta[,"IC50 sample date"]
age<-as.numeric(floor((IC50date-dob)/365))
pretreated<-patmeta[,"Treatment y/n"]
#replace NAs in age by median
#age[is.na(age)]<-median(age)
clinicalData <- cbind(age=age,gender=gender, pretreated=pretreated)
rownames(clinicalData) <- rownames(patmeta)
clinicalData<-addMissingPatients(patientID, clinicalData)
checkNA(as.matrix(clinicalData))
```



##Export
Export Data 
```{r}
# save(DrugBase, DrugAll,CompleteDF,df4ana,SetUp12,SetUpRest, df4anaAvreplicates, patcol,genData, methData, IGHVData, eData, file="DrugCombi_Linkmolecular.RData")
```


## Annotate Barplot and Curves With IGHV

Creates the following plots:

* "DiffPerConc_IGHVann.pdf" -same plot as "DiffPerConc.pdf" with asteriks markin M-CLL patients
* "ABC_IGHVann.pdf" -same plot as "ABC.pdf" with asteriks marking M-CLL patients
* "Curves_IGHVann.pdf" -same plot as "Curves.pdf" but without the curves for every patient instead one curve displaying median of all M-CLL patient (red), one curve displaying median of all M-CLL patient (green), one curve displaying the overall median (black). Again solid lines correspond to the measured viability value with both drugs,  dashed lines to the values from a theoretical additive model. 

```{r, eval=T}
source("PlotThings.R")
BaseDrugs<-unique(df4ana$BDrugName)
CombiDrugs<-unique(df4ana$CDrugAbrv)
plotThings("DiffPerConc", IGHV=IGHVData,CombiDrugs=CombiDrugs,
                     BaseDrugs=BaseDrugs, df4anaAvreplicates=df4anaAvreplicates, outputdir=outputdir)
plotThings("ABC", IGHV=IGHVData,CombiDrugs=CombiDrugs,
                     BaseDrugs=BaseDrugs, df4anaAvreplicates=df4anaAvreplicates, outputdir=outputdir)
plotThings("Curves", IGHV=IGHVData,CombiDrugs=CombiDrugs,
                     BaseDrugs=BaseDrugs, df4anaAvreplicates=df4anaAvreplicates, outputdir=outputdir)
```
In the curves plot the black curve is the overall median per concetration across patient, the red curve is the median for all M-CLL patients, the green curve the median for all U-CLL patients. Dashed and solid are as for the indivivual patients the theoretical additive value and the measured combination value, respectively.


## t-test for mutations and dotplots
Each combination in every concentration is tested for significant difference in viabilitites for certain genetic subgroups, i.e. IGHV, tr12 TP53 using a t-test and adjusting for mutliple testing. Boxplots with dotplots are drawn.
```{r, eval=T}
BaseDrugs<-as.character(unique(df4ana$BDrugName))
CombiDrugs<-c(as.character(unique(df4ana$CDrugAbrv)))#, "DMSO")
concs<-paste("c",1:5, sep="")
muts<-c("IGHV","trisomy12", "TP53","del13q14", "del11q22.3")
ttestDF<-expand.grid(concs,CombiDrugs,BaseDrugs, muts, stringsAsFactors = F)
colnames(ttestDF)<-c("concB","DrugC","DrugB",  "muts")
ttestDF<-cbind(ttestDF, pval=NA, meandiff=NA)
par(mfrow=c(2,3))
for(i in 1:2){
for(nr in 1:nrow(ttestDF)){
    mut<-ttestDF[nr,"muts"]
    DrugB<-ttestDF[nr,"DrugB"]
    concB<-ttestDF[nr,"concB"]
    DrugC<-ttestDF[nr, "DrugC"]

   if(mut=="IGHV") df4anaAvreplicates$mut<-as.factor(IGHVData[as.character(df4anaAvreplicates$PatientID),])
    else df4anaAvreplicates$mut<-as.factor(genData[as.character(df4anaAvreplicates$PatientID),mut])
  #if DMSO chose arbitrary C drug and use effectB column
  
  DMSObool<-F
  if(DrugC=="DMSO") {DrugC<-"Ibrutinib100"; DMSObool<-T}
  
  df4focus<-filter(df4anaAvreplicates, BDrugName==DrugB & CDrugAbrv==DrugC)
  
  if(!DMSObool) df4focus$effectVal<-df4focus$effectBC
  if(DMSObool) {df4focus$effectVal<-df4focus$effectB; df4focus$CDrugAbrv=="DMSO"; DrugC<-"DMSO"}
  
    maxy<-max(df4focus$effectVal)+0.1
    df4focus<-filter(df4focus, BDrugConcId==concB &!is.na(mut))

  if(i==1){
  # two-sided t test with equal variance on combination effect between the two groups if at leat 3 patients per group
  if(sum(df4focus$mut==0)>2 &sum(df4focus$mut==1)>2){
    ttestout<-t.test(df4focus$effectVal ~ df4focus$mut, var.equal=TRUE)
    ttestDF$pval[nr]<-ttestout$p.value
    ttestDF$meandiff[nr]<-ttestout$estimate[2]-ttestout$estimate[1]
  }
  else {ttestDF$pval[nr]<-NA; ttestDF$meandiff[nr]<-NA
  }
  }
  
  if(i==2){
    #correct for multiple testing and plot
    if(nr==1) {ttestDF$pval.adj<-p.adjust(ttestDF$pval, "BH"); hist(ttestDF$pval, breaks=seq(0,1,0.05))}
    
    if(concB=="c1") {
      if(!file.exists(file.path(outputdir,"dotplots"))) dir.create(file.path(outputdir,"dotplots"))
      pdf(file.path(outputdir,"dotplots",paste(DrugB,"_" ,DrugC,"_" ,mut,".pdf", sep="")))
      par(mfrow=c(2,3),oma=c(0,0,2,0))
    }
    plot(NA, xlim=c(0.5,2.5),ylim=c(0,maxy), xlab="", 
        ylab=paste("viability "), xaxt="n", yaxt="n", yaxs="i", main="")
    boxplot(df4focus$effectVal ~ df4focus$mut,outline=FALSE, add=TRUE, boxwex=0.3, xaxt="n", yaxt="n")
    beeswarm(df4focus$effectVal ~ df4focus$mut, corral="wrap", add=TRUE, col=adjustcolor("sienna3", alpha.f = 0.5) , pch=20)
    axis(side=1, at=c(1,2), labels=NA)
    mtext(side=1, at=c(1,2), text=c("wt", "mt"), cex=1, line=0.3)
    mtext(side=1, at=c(1.5), text=mut, cex=1, line=1.5)
    axis(side=2, at=seq(0,maxy,0.2), labels=seq(0,maxy,0.2), las=2)
    mtext(side=3, at=c(1.5), text=paste("p.adj=", round(ttestDF[nr, "pval.adj"],4), "p=", round(ttestDF[nr, "pval"],4)), cex=1, line=1)
    mtext(side=3, at=c(1.5), text=paste("conc=", concB), cex=1, line=2.5)


    if(concB=="c5") {title(paste("C=", DrugC,", B=",DrugB), outer=TRUE); dev.off()}
    }

}
}

save(ttestDF, file=file.path(outputdir,"ttestDFCombi.RData"))
#load("ttestDFCombi.RData")
write.csv(ttestDF, file=file.path(outputdir,"pvalues_viabilities.csv"))
```

Consider significant associations: Write to a table containing all base drugs (not DM as base) with a combinatory sbstance that shows a significant association to one of IGHV, TP53 or tr12 at FDR 10%.
```{r, eval=T}
sigDF<-filter(ttestDF, pval.adj<0.1)
head(sigDF)
highlight<-select(sigDF, DrugB, DrugC, muts)
highlight<-highlight[!duplicated(highlight),]
rownames(highlight)<-NULL

write.csv(highlight, file.path(outputdir,"SignAssocForBaseDrugsFDR10.csv"))
```


Produce the plot "volcanoeAssoc.pdf" showing the p-values for association from viability values of drug combinations to the genetic features given by TP53, Trisomy12 and IGHV.

```{r, eval=T, echo=F}
#function of Gosia
ggvolc = function(df, title, Ycut, color=c("deeppink", "navy"), xlab) {
  
  # check if dataq.frame have required columns
  stopifnot(all(c("X","Y","Label","Grey") %in% colnames(df)))
  
  # check if colors are for the palette # if not use the default
  col.default = c("deeppink", "navy")
  color = ifelse(color %in% colors(), color,  col.default)
  
  # function for colors
  col2hex = function(cols, alpha=1, names=NA) {
    tmp = col2rgb(cols) 
    max=255
    tmp = apply(tmp,2, function(t) rgb(red=t[1], green=t[2], blue=t[3], maxColorValue=max, alpha=alpha*max))
    if(all(!is.na(names)) & length(names)==length(tmp)) tmp = setNames(tmp,nm=names)
    tmp
  }

  # xlim
  minX = min(df$X)
  maxX = max(df$X)
  
  # y axis labels
  maxY = max(ceiling(max(df$Y)), 4)  # minimum ylim is set to 4
  axisMark =if(maxY<5) 1 else if(maxY<10) 2 else if(maxY<15) 4 else 5
  
  # direction of the effect
  df$Direction = 0
  df$Direction[df$Y>=Ycut & df$X<0] = -1
  df$Direction[df$Y>=Ycut & df$X>0] = 1
  df$Direction[df$Y>=Ycut & df$Grey] = 2
  df$Direction = factor(df$Direction, levels=c(-1,1,0,2))
  
   # add column saying if association is significant
  df$IsSignificant = df$Y>=Ycut & !df$Grey
  
  ## ONLY IF THERE IS ANY SIGNIFICANT RESULT
  if(sum(df$IsSignificant)>0) {
    
    # hanging ends for labels
    hend = max(abs(c(minX, maxX)))*0.7 #0.42
    
    # shift of labels and lines
    shiftX = max(abs(c(minX, maxX)))*0.05
    
    # positions for labels
    calcY = function(y.org) {
      rng = range(y.org)
      inc = max((rng[2]-rng[1])/length(y.org), 0.1)
      newY = rng[1]+inc*(1:length(y.org))
      y.org[order(y.org)] = newY
      y.org
    }
    df$labY[df$IsSignificant] = calcY(y.org=df$Y[df$IsSignificant])
    df$labX = with(df, ifelse(IsSignificant, ifelse(Direction==1, maxX+shiftX, minX-shiftX), NA))
    df$hjust = ifelse(sign(df$labX)==1, 0, 1)
    
  } else {
    # hanging ends for labels
    hend = 0
  }
  
  gg = ggplot() + geom_point(data=df, aes(x=X, y=Y, fill=Direction, colour=Direction), size=ifelse(df$IsSignificant,3,2), shape=21) + scale_colour_manual(values=c(`-1`=color[1], `1`=color[2], `0`="black", `2`="darkgrey"), guide=FALSE) + scale_fill_manual(values=c(col2hex(c(color[1], color[2]), names=c("-1","1"), alpha=0.7), "0"="black", `2`="grey"), guide=FALSE) + theme_bw() + geom_hline(yintercept=Ycut, colour="navy", linetype="dashed") + geom_vline(xintercept=0, colour="navy", size=0.1) + ylab("") + xlab(xlab) + ggtitle(title) + scale_y_continuous("P-value", breaks=seq(1,maxY,axisMark), labels=10^(-seq(1,maxY,axisMark)), limits=c(0,maxY)) + xlim(minX-hend, maxX+hend)
  
   if(sum(df$IsSignificant)>0) {
     df = df[df$IsSignificant,]
     gg = gg + geom_segment(data=df, aes(x=X, y=Y, xend=labX, yend=labY), colour="darkgrey", alpha=0.7, linetype="dotted") + geom_text(data=df, mapping=aes(x=labX, y=labY, label=Label, colour=Direction, hjust=hjust), size=2.5)
   }
  gg
}
```


```{r, eval=T}
pdf(file.path(outputdir,"volcanoeAssoc.pdf"))
for(mut2plot in c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")){
dftmp<-filter(ttestDF, muts==mut2plot)
#use only one concentratio per combination (the one with lowest p-value)
dfnoConc<-dftmp[seq(1, nrow(dftmp),5),]
dfnoConc$which<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, i+which.min(dftmp$pval.adj[i:(i+4)])-1))
dfnoConc$howmany<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, sum(dftmp$pval.adj[i:(i+4)]<0.1)))
dfnoConc<-filter(dfnoConc, !is.na(which))
dfnoConc$pval.adj<-dftmp$pval.adj[dfnoConc$which]
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$meandiff<-dftmp$meandiff[dfnoConc$which]

df4volc<-data.frame(X=dfnoConc$meandiff, Y=-log10(dfnoConc$pval.adj), 
                    Label=paste(dfnoConc$DrugB,"+" ,dfnoConc$DrugC), Grey=F)
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot, "(most signif. conc)"), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```


```{r, eval=T}
pdf(file.path(outputdir,"volcanoeAssocCombiOnly.pdf"))
for(mut2plot in c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")){
dftmp<-filter(ttestDF, muts==mut2plot)
#use only one concentratio per combination (the one with lowest p-value)
dfnoConc<-dftmp[seq(1, nrow(dftmp),5),]
dfnoConc$which<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, i+which.min(dftmp$pval.adj[i:(i+4)])-1))
dfnoConc$howmany<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, sum(dftmp$pval.adj[i:(i+4)]<0.1)))
dfnoConc<-filter(dfnoConc, !is.na(which))
dfnoConc$pval.adj<-dftmp$pval.adj[dfnoConc$which]
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$meandiff<-dftmp$meandiff[dfnoConc$which]

df4volc<-data.frame(X=dfnoConc$meandiff, Y=-log10(dfnoConc$pval.adj), 
                    Label=paste(dfnoConc$DrugB,"+" ,dfnoConc$DrugC), Grey=F)
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot, "(most signif. conc)"), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```


##Volcanoe Plot an t-test for only base durgs
```{r, eval=T}
BaseDrugs<-as.character(unique(df4ana$BDrugName))
CombiDrugs<-c("DMSO")
concs<-paste("c",1:5, sep="")
muts<-c("IGHV","trisomy12", "TP53","del13q14", "del11q22.3")
ttestDFbase<-expand.grid(concs,CombiDrugs,BaseDrugs, muts, stringsAsFactors = F)
colnames(ttestDFbase)<-c("concB","DrugC","DrugB",  "muts")
ttestDFbase<-cbind(ttestDFbase, pval=NA, meandiff=NA)
par(mfrow=c(2,3))
for(i in 1:2){
for(nr in 1:nrow(ttestDFbase)){
    mut<-ttestDFbase[nr,"muts"]
    DrugB<-ttestDFbase[nr,"DrugB"]
    concB<-ttestDFbase[nr,"concB"]
    DrugC<-ttestDFbase[nr, "DrugC"]

   if(mut=="IGHV") df4anaAvreplicates$mut<-as.factor(IGHVData[as.character(df4anaAvreplicates$PatientID),])
    else df4anaAvreplicates$mut<-as.factor(genData[as.character(df4anaAvreplicates$PatientID),mut])
  #if DMSO chose arbitrary C drug and use effectB column
  
  DMSObool<-F
  if(DrugC=="DMSO") {DrugC<-"Ibrutinib100"; DMSObool<-T}
  
  df4focus<-filter(df4anaAvreplicates, BDrugName==DrugB & CDrugAbrv==DrugC)
  
  if(!DMSObool) df4focus$effectVal<-df4focus$effectBC
  if(DMSObool) {df4focus$effectVal<-df4focus$effectB; df4focus$CDrugAbrv=="DMSO"; DrugC<-"DMSO"}
  
    maxy<-max(df4focus$effectVal)+0.1
    df4focus<-filter(df4focus, BDrugConcId==concB &!is.na(mut))

  if(i==1){
  # two-sided t test with equal variance on combination effect between the two groups if at leat 3 patients per group
  if(sum(df4focus$mut==0)>2 &sum(df4focus$mut==1)>2){
    ttestout<-t.test(df4focus$effectVal ~ df4focus$mut, var.equal=TRUE)
    ttestDFbase$pval[nr]<-ttestout$p.value
    ttestDFbase$meandiff[nr]<-ttestout$estimate[2]-ttestout$estimate[1]
  }
  else {ttestDFbase$pval[nr]<-NA; ttestDFbase$meandiff[nr]<-NA
  }
  }
  
  if(i==2){
    #correct for multiple testing and plot
    if(nr==1) {ttestDFbase$pval.adj<-p.adjust(ttestDFbase$pval, "BH"); hist(ttestDFbase$pval, breaks=seq(0,1,0.05))}
    
    if(concB=="c1") {
      if(!file.exists(file.path(outputdir,"dotplots"))) dir.create(file.path(outputdir,"dotplots"))
      pdf(file.path(outputdir,"dotplots",paste(DrugB,"_" ,DrugC,"_" ,mut,".pdf", sep="")))
      par(mfrow=c(2,3),oma=c(0,0,2,0))
    }
    plot(NA, xlim=c(0.5,2.5),ylim=c(0,maxy), xlab="", 
        ylab=paste("viability "), xaxt="n", yaxt="n", yaxs="i", main="")
    boxplot(df4focus$effectVal ~ df4focus$mut,outline=FALSE, add=TRUE, boxwex=0.3, xaxt="n", yaxt="n")
    beeswarm(df4focus$effectVal ~ df4focus$mut, corral="wrap", add=TRUE, col=adjustcolor("sienna3", alpha.f = 0.5) , pch=20)
    axis(side=1, at=c(1,2), labels=NA)
    mtext(side=1, at=c(1,2), text=c("wt", "mt"), cex=1, line=0.3)
    mtext(side=1, at=c(1.5), text=mut, cex=1, line=1.5)
    axis(side=2, at=seq(0,maxy,0.2), labels=seq(0,maxy,0.2), las=2)
    mtext(side=3, at=c(1.5), text=paste("p.adj=", round(ttestDFbase[nr, "pval.adj"],4), "p=", round(ttestDFbase[nr, "pval"],4)), cex=1, line=1)
    mtext(side=3, at=c(1.5), text=paste("conc=", concB), cex=1, line=2.5)


    if(concB=="c5") {title(paste("C=", DrugC,", B=",DrugB), outer=TRUE); dev.off()}
    }

}
}

save(ttestDFbase, file=file.path(outputdir,"ttestDFbaseonly.RData"))
#load("ttestDFbaseonly.RData")
write.csv(ttestDFbase, file=file.path(outputdir,"pvalues_viabilities_baseonly.csv"))
```


Produce the plot "volcanoeAssocBasOnyl.pdf" showing the p-values for association from viability values of drug combinations to the genetic features given by TP53, Trisomy12 and IGHV.
```{r, eval=T}
pdf(file.path(outputdir,"volcanoeAssocBaseOnly.pdf"))
for(mut2plot in c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")){
dftmp<-filter(ttestDFbase, muts==mut2plot)
#use only one concentratio per combination (the one with lowest p-value)
dfnoConc<-dftmp[seq(1, nrow(dftmp),5),]
dfnoConc$which<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, i+which.min(dftmp$pval.adj[i:(i+4)])-1))
dfnoConc$howmany<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, sum(dftmp$pval.adj[i:(i+4)]<0.1)))
dfnoConc<-filter(dfnoConc, !is.na(which))
dfnoConc$pval.adj<-dftmp$pval.adj[dfnoConc$which]
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$meandiff<-dftmp$meandiff[dfnoConc$which]

df4volc<-data.frame(X=dfnoConc$meandiff, Y=-log10(dfnoConc$pval.adj), 
                    Label=paste(dfnoConc$DrugB), Grey=F)
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot, "(most signif. conc)"), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```

## t-test for for synergy at each concentration
Each combination in every concentration is tested for significant difference in synergy index (from additive model) for certain genetic subgroups, i.e. IGHV, tr12 TP53 using a t-test and adjusting for mutliple testing. Boxplots with dotplots are drawn. 
```{r, eval=T}
load("df4CI.RData")
df4CImut<-mutate(df4CI, 
                 IGHV=IGHVData[as.character(PatientID),], 
                 TP53=genData[as.character(PatientID),"TP53"],
                 trisomy12=genData[as.character(PatientID),"trisomy12"],
                 del13q14=genData[as.character(PatientID),"del13q14"],
                 del11q22.3=genData[as.character(PatientID),"del11q22.3"])

BaseDrugs<-as.character(unique(df4CImut$BDrugName))
CombiDrugs<-as.character(unique(df4CImut$CDrugAbrv))
concs<-paste("c",1:5, sep="")
muts<-c("IGHV","trisomy12", "TP53","del13q14", "del11q22.3")
ttestDFsyn<-expand.grid(concs,CombiDrugs,BaseDrugs, muts, stringsAsFactors = F)
colnames(ttestDFsyn)<-c("concB","DrugC","DrugB",  "muts")
ttestDFsyn<-cbind(ttestDFsyn, pval=NA, meandiff=NA)
par(mfrow=c(2,3))
for(i in 1:2){
for(nr in 1:nrow(ttestDFsyn)){
    mut<-ttestDFsyn[nr,"muts"]
    DrugB<-ttestDFsyn[nr,"DrugB"]
    concB<-ttestDFsyn[nr,"concB"]
    DrugC<-ttestDFsyn[nr, "DrugC"]

  df4focus<-filter(df4CImut, BDrugName==DrugB & CDrugAbrv==DrugC)
  
  df4focus$effectVal<-df4focus$DiffAddSI
  df4focus$mut<-df4focus[,mut]
    maxy<-round(max(df4focus$effectVal, na.rm = T),1)+0.1
        miny<-round(min(df4focus$effectVal, na.rm = T),1)-0.1

    df4focus<-filter(df4focus, BDrugConcId==concB &!is.na(mut))

  if(i==1){
  # two-sided t test with equal variance on combination effect between the two groups if at leat 3 patients per group
  if(sum(df4focus$mut==0)>2 &sum(df4focus$mut==1)>2){
    ttestout<-t.test(df4focus$effectVal ~ df4focus$mut, var.equal=TRUE)
    ttestDFsyn$pval[nr]<-ttestout$p.value
    ttestDFsyn$meandiff[nr]<-ttestout$estimate[2]-ttestout$estimate[1]
  }
  else {ttestDFsyn$pval[nr]<-NA; ttestDFsyn$meandiff[nr]<-NA
  }
  }
  
  if(i==2){
    #correct for multiple testing and plot
    if(nr==1) {
      ttestDFsyn$pval.adj<-p.adjust(ttestDFsyn$pval, "BH")
      hist(ttestDFsyn$pval, breaks=seq(0,1,0.05))
      ggplot(ttestDFsyn)+geom_histogram(aes(x=pval, group=DrugC))+facet_wrap(~DrugC)+ggtitle("p-value histograms for synergy testing by combination drug")
      pdf(file.path(outputdir,"p-value_histogramsByC.pdf"))
      gg<- ggplot(ttestDFsyn)+geom_histogram(aes(x=pval, group=DrugC))+facet_wrap(~DrugC)+ggtitle("p-value histograms for synergy testing by combination drug")
      print(gg)
      dev.off()
      }
    
    if(concB=="c1") {
      if(!file.exists(file.path(outputdir,"dotplotsSyn"))) dir.create(file.path(outputdir,"dotplotsSyn"))
      pdf(file.path(outputdir,"dotplotsSyn",paste(DrugB,"_" ,DrugC,"_" ,mut,".pdf", sep="")))
      par(mfrow=c(2,3),oma=c(0,0,2,0))
    }
    plot(NA, xlim=c(0.5,2.5),ylim=c(miny,maxy), xlab="", 
        ylab=paste("difference to additive model "), xaxt="n", yaxt="n", yaxs="i", main="")
    boxplot(df4focus$effectVal ~ df4focus$mut,outline=FALSE, add=TRUE, boxwex=0.3, xaxt="n", yaxt="n")
    beeswarm(df4focus$effectVal ~ df4focus$mut, corral="wrap", add=TRUE, col=adjustcolor("sienna3", alpha.f = 0.5) , pch=20)
    axis(side=1, at=c(1,2), labels=NA)
    mtext(side=1, at=c(1,2), text=c("wt", "mt"), cex=1, line=0.3)
    mtext(side=1, at=c(1.5), text=mut, cex=1, line=1.5)
    axis(side=2, at=seq(miny,maxy,0.2), labels=seq(miny,maxy,0.2), las=2)
    mtext(side=3, at=c(1.5), text=paste("p.adj=", round(ttestDFsyn[nr, "pval.adj"],4), "p=", round(ttestDFsyn[nr, "pval"],4)), cex=1, line=1)
    mtext(side=3, at=c(1.5), text=paste("conc=", concB), cex=1, line=2.5)


    if(concB=="c5") {title(paste("C=", DrugC,", B=",DrugB), outer=TRUE); dev.off()}
    }

}
}

save(ttestDFsyn, file=file.path(outputdir,"ttestDFsyn.RData"))
#load("ttestDFsyn.RData")
write.csv(ttestDFsyn, file=file.path(outputdir,"pvalues_SI.csv"))

```

Produce the plot "volcanoeAssocSyn.pdf" showing the p-values for association from synergy values to the genetic features given above. NO SIGNIFICANCE BUT HEAVY MULTIPLETESTING BURDEN. Maybe use IHW with drug C as grouping, see histograms above no more power exprected
```{r, eval=F}
pdf(file.path(outputdir,"volcanoeAssocSyn.pdf"))
for(mut2plot in c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")){
dftmp<-filter(ttestDFsyn, muts==mut2plot)
#use only one concentratio per combination (the one with lowest p-value)
dfnoConc<-dftmp[seq(1, nrow(dftmp),5),]
dfnoConc$which<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, i+which.min(dftmp$pval.adj[i:(i+4)])-1))
dfnoConc$howmany<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval.adj[i:(i+4)])), NA, sum(dftmp$pval.adj[i:(i+4)]<0.1)))
dfnoConc<-filter(dfnoConc, !is.na(which))
dfnoConc$pval.adj<-dftmp$pval.adj[dfnoConc$which]
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$meandiff<-dftmp$meandiff[dfnoConc$which]

df4volc<-data.frame(X=dfnoConc$meandiff, Y=-log10(dfnoConc$pval.adj), 
                    Label=paste(dfnoConc$DrugB,"+" ,dfnoConc$DrugC), Grey=F)
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot, "(most signif. conc)"), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```

```{r, eval=F}
pdf(file.path(outputdir,"volcanoeAssocSyn_rawP.pdf"))
for(mut2plot in c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")){
dftmp<-filter(ttestDFsyn, muts==mut2plot)
#use only one concentratio per combination (the one with lowest p-value)
dfnoConc<-dftmp[seq(1, nrow(dftmp),5),]
dfnoConc$which<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval[i:(i+4)])), NA, i+which.min(dftmp$pval.adj[i:(i+4)])-1))
dfnoConc$howmany<-sapply(seq(1, nrow(dftmp),5), function(i) ifelse(all(is.na(dftmp$pval[i:(i+4)])), NA, sum(dftmp$pval[i:(i+4)]<0.1)))
dfnoConc<-filter(dfnoConc, !is.na(which))
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$pval<-dftmp$pval[dfnoConc$which]
dfnoConc$meandiff<-dftmp$meandiff[dfnoConc$which]

df4volc<-data.frame(X=dfnoConc$meandiff, Y=-log10(dfnoConc$pval), 
                    Label=paste(dfnoConc$DrugB,"+" ,dfnoConc$DrugC), Grey=F)
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot, "(most signif. conc, raw p)"), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```


#t-test on SI 
```{r, eval=T}
dfAggr<-select(df4anaAvreplicates,CDrugAbrv, BDrugName, PatientID, effectC) 
dfAggr<-dfAggr[!duplicated(dfAggr),]
dfAggr$ABC<-sapply(1:nrow(dfAggr), function(i){
dfidx<-dfAggr[i,] 
subDF<-df4anaAvreplicates  %>%  
  filter(CDrugAbrv==as.character(dfidx$"CDrugAbrv"),
         BDrugName==as.character(dfidx$"BDrugName"),
         PatientID==as.character(dfidx$"PatientID"))
#claculate area between the curves and normalize by highest concentration used
(-trapz(subDF$BDrugConc, subDF$effectB*subDF$effectC)-(-trapz(subDF$BDrugConc, subDF$effectBC)))/as.numeric(filter(DrugBase,Substance==dfidx$"BDrugName")[4]) #minus because concentrations ordered from high to low
}) 

dfAggr<-mutate(dfAggr, 
                 IGHV=IGHVData[as.character(PatientID),], 
                 TP53=genData[as.character(PatientID),"TP53"],
                 trisomy12=genData[as.character(PatientID),"trisomy12"],
                 del13q14=genData[as.character(PatientID),"del13q14"],
                 del11q22.3=genData[as.character(PatientID),"del11q22.3"])

DFmeanABC<-dfAggr%>% group_by(CDrugAbrv,BDrugName) %>% dplyr::summarize( meanABC=mean(ABC),medianABC=median(ABC))
colnames(DFmeanABC)<-c("DrugC", "DrugB", "meanABC", "medianABC")


BaseDrugs<-as.character(unique(dfAggr$BDrugName))
CombiDrugs<-as.character(unique(dfAggr$CDrugAbrv))
muts<-c("IGHV", "TP53","trisomy12","del13q14", "del11q22.3")
ttestDFsyn<-expand.grid(CombiDrugs,BaseDrugs, muts, stringsAsFactors = F)
colnames(ttestDFsyn)<-c("DrugC","DrugB",  "muts")
ttestDFsyn<-cbind(ttestDFsyn, pval=NA, meandiff=NA)

for(i in 1:2){
for(nr in 1:nrow(ttestDFsyn)){
    mut<-ttestDFsyn[nr,"muts"]
    DrugB<-ttestDFsyn[nr,"DrugB"]
    DrugC<-ttestDFsyn[nr, "DrugC"]

  df4focus<-filter(dfAggr, BDrugName==DrugB & CDrugAbrv==DrugC)
  
  df4focus$effectVal<-df4focus$ABC
  df4focus$mut<-df4focus[,mut]
    maxy<-round(max(df4focus$effectVal, na.rm = T),1)+0.1
    miny<-round(min(df4focus$effectVal, na.rm = T),1)-0.1

    df4focus<-filter(df4focus, !is.na(mut))

  if(i==1){
  # two-sided t test with equal variance on combination effect between the two groups if at leat 3 patients per group
  if(sum(df4focus$mut==0)>2 &sum(df4focus$mut==1)>2){
    ttestout<-t.test(df4focus$effectVal ~ df4focus$mut, var.equal=TRUE)
    ttestDFsyn$pval[nr]<-ttestout$p.value
    ttestDFsyn$meandiff[nr]<-ttestout$estimate[2]-ttestout$estimate[1]
  }
  else {ttestDFsyn$pval[nr]<-NA; ttestDFsyn$meandiff[nr]<-NA
  }
  }
  
  if(i==2){
    # correct for multiple testing and plot
    if(nr==1) {
      ttestDFsyn$pval.adj<-p.adjust(ttestDFsyn$pval, "BH")
      hist(ttestDFsyn$pval, breaks=seq(0,1,0.05))
      pdf(file.path(outputdir,"p-value_histogramsByC.pdf"))
      gg<- ggplot(ttestDFsyn)+geom_histogram(aes(x=pval, group=DrugC))+facet_wrap(~DrugC)+ggtitle("p-value histograms for synergy testing by combination drug")
      print(gg)
      dev.off()
      }

  
      if(!file.exists(file.path(outputdir,"dotplotsSyn"))) dir.create(file.path(outputdir,"dotplotsSyn"))
      pdf(file.path(outputdir,"dotplotsSyn",paste(DrugB,"_" ,DrugC,"_" ,mut,".pdf", sep="")))
      par(mfrow=c(2,3),oma=c(0,0,2,0))
    
      plot(NA, xlim=c(0.5,2.5),ylim=c(miny,maxy), xlab="", 
      ylab=paste("difference to additive model "), xaxt="n", yaxt="n", yaxs="i", main="")
    boxplot(df4focus$effectVal ~ df4focus$mut,outline=FALSE, add=TRUE, boxwex=0.3, xaxt="n", yaxt="n")
    beeswarm(df4focus$effectVal ~ df4focus$mut, corral="wrap", add=TRUE, col=adjustcolor("sienna3", alpha.f = 0.5) , pch=20)
    axis(side=1, at=c(1,2), labels=NA)
    mtext(side=1, at=c(1,2), text=c("wt", "mt"), cex=1, line=0.3)
    mtext(side=1, at=c(1.5), text=mut, cex=1, line=1.5)
    axis(side=2, at=seq(miny,maxy,0.2), labels=seq(miny,maxy,0.2), las=2)
    mtext(side=3, at=c(1.5), text=paste("p.adj=", round(ttestDFsyn[nr, "pval.adj"],4), "p=", round(ttestDFsyn[nr, "pval"],4)), cex=1, line=1)
    title(paste("C=", DrugC,", B=",DrugB), outer=TRUE) 
    dev.off()
    }

}
}

save(ttestDFsyn, file=file.path(outputdir,"ttestDF_SI.RData"))
#load("ttestDFABC.RData")
write.csv(ttestDFsyn, file=file.path(outputdir,"pvalues_SI.csv"))
```


After multiple testin correction no results are significant. This is also the case when testing only highly synergestic combinations or a subset of mutations. There seems to be no different p-value historgrams dpednign on combiniation drug.
```{r}
ttestDFsyn<-ttestDFsyn[!is.na(ttestDFsyn$pval),]
ttestDFsyn$pvalBH<-p.adjust(ttestDFsyn$pval, "BH")
ttestDFsyn<-merge(ttestDFsyn, DFmeanABC, by=c("DrugC", "DrugB"))

ttestDFsyn$pvalBH<-p.adjust(ttestDFsyn$pval, "BH")

pdf(file.path(outputdir,"volcanoeAssocABC.pdf"))
for(mut2plot in c("IGHV", "TP53")){ #c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")
dftmp<-filter(ttestDFsyn, muts==mut2plot)
df4volc<-data.frame(X=dftmp$meandiff, Y=-log10(dftmp$pvalBH), 
                    Label=paste(dftmp$DrugB,"+" ,dftmp$DrugC), Grey=F)
df4volc<-df4volc[!is.na(df4volc$Y),]
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()


pdf(file.path(outputdir,"volcanoeAssocSI_rawP.pdf"))
for(mut2plot in c("IGHV", "TP53")){ #c("IGHV", "trisomy12", "TP53" ,"del11q22.3", "del13q14")
dftmp<-filter(ttestDFsyn, muts==mut2plot)
df4volc<-data.frame(X=dftmp$meandiff, Y=-log10(dftmp$pval), 
                    Label=paste(dftmp$DrugB,"+" ,dftmp$DrugC), Grey=F)
df4volc<-df4volc[!is.na(df4volc$Y),]
gg<-ggvolc(df4volc, title=paste("Volcanoe Plot for associations with", mut2plot), Ycut=-log10(0.05), xlab="Mean effect difference in mutated and unmutated patients")
print(gg)
}
dev.off()
```



#Influence of TP53 mt and IGHV on Synergy of Fludarabine and Ibrutinib

##Boxplots
```{r}
#0 both unmut, 1 TP53mt UCLL, 2
df4CImut$IGHV_TP53<-2*df4CImut$IGHV+df4CImut$TP53
label<-c('0'="TP53wt UCLL", '1'="TP53mt UCLL",'2'="TP53wt MCLL",'3'="TP53mt MCLL")
df4CImut$IGHV_TP53<-label[as.character(df4CImut$IGHV_TP53)]
df4CImutFluIbru100<-df4CImut%>%filter(BDrugName=="Fludarabine", CDrugAbrv=="Ibrutinib100")
ggplot(df4CImutFluIbru100, aes(y=DiffAddSI, x=IGHV_TP53,group=IGHV_TP53, fill=IGHV_TP53))+geom_boxplot()+ggtitle("Synergy Flu +Ibru100")+facet_wrap(~BDrugConcId, scales = "free")

#get rid of outliers
head(df4CImut)
df4CImutRmOut<-df4CImutFluIbru100%>% filter( viabBC<1.4 & viabB<1.4 & viabC< 1.4)
ggplot(df4CImutRmOut, aes(y=DiffAddSI, x=IGHV_TP53,group=IGHV_TP53, fill=IGHV_TP53))+geom_boxplot()+ggtitle("Synergy Flu +Ibru100 (after filtering out viabilites>1.4)")+facet_wrap(~BDrugConcId)


#on AUC
AUC<-sapply(unique(df4CImutFluIbru100$PatientID), function (pat){
subDF<-df4CImutFluIbru100  %>%  filter(PatientID==pat)
#claculate area between the curves and normalize by highest concentration used
(-trapz(subDF$BDrugConc, subDF$viabB*subDF$viabC)-(-trapz(subDF$BDrugConc, subDF$viabBC)))/as.numeric(filter(DrugBase,Substance=="Fludarabine")[4]) #minus because concentrations ordered from high to low
}) 
dfAUC<-data.frame(SI=AUC, PatientID=unique(df4CImutFluIbru100$PatientID))
dfAUC$IGHV<-IGHVData[as.character(dfAUC$PatientID), ]
dfAUC$TP53<-genData[as.character(dfAUC$PatientID), "TP53"]
dfAUC$IGHV_TP53<-2*dfAUC$IGHV+dfAUC$TP53
label<-c('0'="TP53wt UCLL", '1'="TP53mt UCLL",'2'="TP53wt MCLL",'3'="TP53mt MCLL")
dfAUC$IGHV_TP53<-label[as.character(dfAUC$IGHV_TP53)]
pdf(file.path(outputdir,"Flu+IbruBoxplots.pdf"))
ggplot(dfAUC, aes(x=IGHV_TP53, y=SI, fill=IGHV_TP53))+geom_violin()+geom_hline(aes(yintercept=median(dfAUC$SI), color="median"))
ggplot(dfAUC, aes(x=IGHV_TP53, y=SI, fill=IGHV_TP53))+geom_boxplot()+geom_hline(aes(yintercept=median(dfAUC$SI), color="median"))
dev.off()
```

 wilcox.test(filter(dfAUC, IGHV_TP53=="TP53wt MCLL")$SI,filter(dfAUC, IGHV_TP53=="TP53wt UCLL")$SI)

	Wilcoxon rank sum test

data:  filter(dfAUC, IGHV_TP53 == "TP53wt MCLL")$SI and filter(dfAUC, IGHV_TP53 == "TP53wt UCLL")$SI
W = 110, p-value = 0.16
alternative hypothesis: true location shift is not equal to 0

> wilcox.test(filter(dfAUC, IGHV_TP53=="TP53mt MCLL")$SI,filter(dfAUC, IGHV_TP53=="TP53mt UCLL")$SI)

	Wilcoxon rank sum test

data:  filter(dfAUC, IGHV_TP53 == "TP53mt MCLL")$SI and filter(dfAUC, IGHV_TP53 == "TP53mt UCLL")$SI
W = 6, p-value = 0.1091
alternative hypothesis: true location shift is not equal to 0

> wilcox.test(filter(dfAUC, IGHV_TP53=="TP53wt MCLL")$SI,filter(dfAUC, IGHV_TP53=="TP53mt MCLL")$SI)

	Wilcoxon rank sum test

data:  filter(dfAUC, IGHV_TP53 == "TP53wt MCLL")$SI and filter(dfAUC, IGHV_TP53 == "TP53mt MCLL")$SI
W = 57, p-value = 0.3892
alternative hypothesis: true location shift is not equal to 0

> wilcox.test(filter(dfAUC, IGHV_TP53=="TP53wt UCLL")$SI,filter(dfAUC, IGHV_TP53=="TP53mt UCLL")$SI)

	Wilcoxon rank sum test

data:  filter(dfAUC, IGHV_TP53 == "TP53wt UCLL")$SI and filter(dfAUC, IGHV_TP53 == "TP53mt UCLL")$SI
W = 48, p-value = 0.6163
alternative hypothesis: true location shift is not equal to 0


#Patient by Drug Heatmap for Combinations
Produces "HeatmapDrugxPatients.pdf" - a heatmap for each combination drug, in which the viability values of the combined drug therapy are displayed and the patients are annotated by certain genetic features. White values in the matrix have a normalized viability value of more than 1.4.
```{r}
heights4Fig<-c(2,1,10,3)
widths4Fig<-c(1,25,1,1,5,1)
pdf(file.path(outputdir, "HeatmapDrugxPatients.pdf"), width=sum(widths4Fig), heigh=sum(heights4Fig))
for(drC in c("DMSO",as.character(unique(df4anaAvreplicates$CDrugAbrv)))){
  bool_DMSO<-F
  if(drC=="DMSO")  bool_DMSO<-T
  
  if(bool_DMSO) dfDrugC<-df4anaAvreplicates else dfDrugC<-filter(df4anaAvreplicates, CDrugAbrv==drC)
if(bool_DMSO)   dftmp<- select(dfDrugC,effectB, PatientID) else dftmp<- select(dfDrugC,effectBC, PatientID)

dftmp$Base<-paste(dfDrugC$BDrugName,"_",
                   dfDrugC$BDrugConcId, sep="")
if(bool_DMSO) dftmp<-dftmp[!duplicated(dftmp),]

PatbyDrugMatrix<-matrix(NA, nrow=length(unique(dftmp$PatientID)),
                        ncol=length(unique(dftmp$Base)))
colnames(PatbyDrugMatrix)<-unique(dftmp$Base)
rownames(PatbyDrugMatrix)<-as.character(unique(dftmp$PatientID))
for(d in 1:nrow(dftmp)){
  tmp<-dftmp[d,]
  if(bool_DMSO) PatbyDrugMatrix[as.character(tmp$PatientID), tmp$Base]<-tmp$effectB else
  PatbyDrugMatrix[as.character(tmp$PatientID), tmp$Base]<-tmp$effectBC
}
#remove Na (patients being outlier in at least one screen)
noOutl<-apply(PatbyDrugMatrix,1,function(r) !any(is.na(r)))
PatbyDrugMatrix<-PatbyDrugMatrix[noOutl,]

colors<- colorRampPalette(c("red", "yellow", "white", "blue"))
callbackPC = function(hc, mat){
    sv = svd(t(mat))$v[,1]
    dend = reorder(as.dendrogram(hc), wts = sv)
    as.hclust(dend)
}

ghm<-pheatmap(PatbyDrugMatrix,
              color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(140),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("viabilty values for combination with", drC),
              breaks=seq(0,1.4,0.01),
              silent=T, fontsize = 18, fontsize_col = 10,fontsize_row = 16, clustering_callback=callbackPC)

#annotate with genetic background
patsInOrder<-ghm$tree_row$labels[ghm$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnno<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=0, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnno<-ggplotGrob(ggMutAnno)


gt<-gtable(width<-unit(widths4Fig, "in"), heights = unit(heights4Fig,"in"))
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[1]], 1, 2) #main
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[2]], 2, 2) #dendrogram columns
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[3]], 3, 1) #dendrogram rows
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[4]], 3, 2) #matrix
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[5]], 4, 2) #column names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[6]], 3, 3) #row names
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[4]], 3, 5) #mutation matrix
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[5]], 4, 5) #mutation names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[7]], 3, 4, clip="off") #legend


grid.draw(gt)
grid.newpage()
}
dev.off()
```

Heatmap based on centred and scaled columns using correlation distance: "HeatmapDrugxPatients_zscore.pdf"
```{r, eval=F}
heights4Fig<-c(2,1,10,3)
widths4Fig<-c(1,25,1,1,5,1)
pdf(file.path(outputdir, "HeatmapDrugxPatients_zscore.pdf"), width=sum(widths4Fig), heigh=sum(heights4Fig))
for(drC in as.character(unique(df4anaAvreplicates$CDrugAbrv))){
dfDrugC<-filter(df4anaAvreplicates, CDrugAbrv==drC)
 dftmp<- select(dfDrugC,effectBC, PatientID)

dftmp$Base<-paste(dfDrugC$BDrugName,"_",
                   dfDrugC$BDrugConcId, sep="")

PatbyDrugMatrix<-matrix(NA, nrow=length(unique(dftmp$PatientID)),
                        ncol=length(unique(dftmp$Base)))
colnames(PatbyDrugMatrix)<-unique(dftmp$Base)
rownames(PatbyDrugMatrix)<-as.character(unique(dftmp$PatientID))
for(d in 1:nrow(dftmp)){
  tmp<-dftmp[d,]
  PatbyDrugMatrix[as.character(tmp$PatientID), tmp$Base]<-tmp$effectBC
}
#remove Na (patients being outlier in at least one screen)
noOutl<-apply(PatbyDrugMatrix,1,function(r) !any(is.na(r)))
PatbyDrugMatrix<-PatbyDrugMatrix[noOutl,]

PatbyDrugMatrix<-apply(PatbyDrugMatrix,2,scale)
rownames(PatbyDrugMatrix)<-as.character(unique(dftmp$PatientID))
colors<- colorRampPalette(c("red", "yellow", "white", "blue"))
ghm<-pheatmap(PatbyDrugMatrix,
              color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(600),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("z-score of viabilty values for combination with", drC),
              breaks=seq(-3,3,0.01),
              silent=T, fontsize = 18, fontsize_col = 10,fontsize_row = 16)

#annotate with genetic background
patsInOrder<-ghm$tree_row$labels[ghm$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnno<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=0, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnno<-ggplotGrob(ggMutAnno)


gt<-gtable(width<-unit(widths4Fig, "in"), heights = unit(heights4Fig,"in"))
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[1]], 1, 2) #main
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[2]], 2, 2) #dendrogram columns
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[3]], 3, 1) #dendrogram rows
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[4]], 3, 2) #matrix
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[5]], 4, 2) #column names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[6]], 3, 3) #row names
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[4]], 3, 5) #mutation matrix
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[5]], 4, 5) #mutation names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[7]], 3, 4, clip="off") #legend


grid.draw(gt)
grid.newpage()
}
dev.off()
```

#Heatmaps Ibrutinib/Idealisib (only common patients)

```{r, eval=F}
df.x<-filter(df4anaAvreplicates, CDrugAbrv=="Ibrutinib100")
df.y<-filter(df4anaAvreplicates, CDrugAbrv=="CAL101")
df<-merge(df.x, df.y, by=c("PatientID","BDrugID", "BDrugConcId",
                                               "BDrugConc", "BDrugName"))
df$BDrug<-paste(df$BDrugName, df$BDrugConcId, sep="_")

dfIbru<-select(df, BDrug, effectBC.x, PatientID)
df4melt<-select(df,PatientID,BDrug, effectBC.x )
matIbru<-melt(df4melt,id.vars=c("PatientID", "BDrug"), measure.vars = "effectBC.x") %>%
  acast(PatientID~BDrug)

dfIdea<-select(df, BDrug, effectBC.y, PatientID)
df4melt<-select(df,PatientID,BDrug, effectBC.y )
matIdea<-melt(df4melt,id.vars=c("PatientID", "BDrug"), measure.vars = "effectBC.y") %>%
  acast(PatientID~BDrug)

#remove Na (patients being outlier in at least one screen)
noOutl<-apply(matIbru,1,function(r) !any(is.na(r))) &apply(matIdea,1,function(r) !any(is.na(r)))
matIbru<-matIbru[noOutl,]
matIdea<-matIdea[noOutl,]

pdf(file.path(outputdir,"HeatmapPatbyDrug_IbruVsIdea_annot.pdf"), width=20, height = 15)
hmIbru<-pheatmap(matIbru, color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(140),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("viabilty values for combination with Ibrutinib100"),
              breaks=seq(0,1.4,0.01),
              silent=F, fontsize = 18, fontsize_col = 10,fontsize_row = 16)

hmIdea<-pheatmap(matIdea, color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(140),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("viabilty values for combination with CAL-101"),
              breaks=seq(0,1.4,0.01),
              silent=F, fontsize = 18, fontsize_col = 10,fontsize_row = 16)

#annotate with genetic background
patsInOrder<-hmIbru$tree_row$labels[hmIbru$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnnoIbru<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=0, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnnoIbru<-ggplotGrob(ggMutAnnoIbru)

patsInOrder<-hmIdea$tree_row$labels[hmIdea$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnnoIdea<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=0, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnnoIdea<-ggplotGrob(ggMutAnnoIdea)


for(drug4hm in c("Idea", "Ibru")){
  if(drug4hm=="Idea") {grobMutAnno<-grobMutAnnoIdea; ghm<- hmIdea
  } else {grobMutAnno<-grobMutAnnoIbru; ghm<- hmIbru}
gt<-gtable(width<-unit(widths4Fig, "in"), heights = unit(heights4Fig,"in"))
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[1]], 1, 2) #main
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[2]], 2, 2) #dendrogram columns
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[3]], 3, 1) #dendrogram rows
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[4]], 3, 2) #matrix
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[5]], 4, 2) #column names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[6]], 3, 3) #row names
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[4]], 3, 5) #mutation matrix
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[5]], 4, 5) #mutation names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[7]], 3, 4, clip="off") #legend


grid.draw(gt)
grid.newpage()
}
dev.off()
```

#Heatmaps on subgroup of patients defined by TP53 or IGHV

```{r, echo=F}
PlotHeatmapIbruIdea4Subgroup<-function( feature="IGHV", status=0){
if(feature=="IGHV") key<-IGHVData else key<-genData[,feature, drop=F] 
key<-key[!is.na(key), ,drop=F]
dfFocus<-filter(df4anaAvreplicates, PatientID %in% rownames(key)[which(key==status)])

df.x<-filter(dfFocus, CDrugAbrv=="Ibrutinib100")
df.y<-filter(dfFocus, CDrugAbrv=="CAL101")
df<-merge(df.x, df.y, by=c("PatientID","BDrugID", "BDrugConcId",
                                               "BDrugConc", "BDrugName"))
df$BDrug<-paste(df$BDrugName, df$BDrugConcId, sep="_")
df$color<-patcol[df$PatientID]

dfIbru<-select(df, BDrug, effectBC.x, PatientID)
df4melt<-select(df,PatientID,BDrug, effectBC.x )
matIbru<-melt(df4melt,id.vars=c("PatientID", "BDrug"), measure.vars = "effectBC.x") %>%
  acast(PatientID~BDrug)

dfIdea<-select(df, BDrug, effectBC.y, PatientID)
df4melt<-select(df,PatientID,BDrug, effectBC.y )
matIdea<-melt(df4melt,id.vars=c("PatientID", "BDrug"), measure.vars = "effectBC.y") %>%
  acast(PatientID~BDrug)

#remove Na (patients being outlier in at least one screen)
noOutl<-apply(matIbru,1,function(r) !any(is.na(r))) &apply(matIdea,1,function(r) !any(is.na(r)))
matIbru<-matIbru[noOutl,]
matIdea<-matIdea[noOutl,]

pdf(file.path(outputdir,paste("HeatmapPatbyDrug_IbruVsIdea", feature, status,".pdf", sep="_")), width=20, height = 15)
pheatmap(matIbru, color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(140),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("viabilty values for combination with Ibrutinib100"),
              breaks=seq(0,1.4,0.01),
              silent=F, fontsize = 18, fontsize_col = 10,fontsize_row = 16)

pheatmap(matIdea, color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(140),
              clustering_distance_rows="correlation", 
              clustering_distance_cols="correlation",
              main=paste("viabilty values for combination with CAL-101"),
              breaks=seq(0,1.4,0.01),
              silent=F, fontsize = 18, fontsize_col = 10,fontsize_row = 16)
dev.off()

pdf(file.path(outputdir,paste("Boxplot_IbruVsIdea", feature, status,".pdf", sep="_")), width=15, height = 20)

#truncate outliers

ggIbru<-ggplot(df )+
  geom_point(aes(x=reorder(BDrug,effectBC.x, median), y=effectBC.x, col=PatientID), alpha=0.7)+
  geom_boxplot(aes(x=reorder(BDrug,effectBC.x, median), y=effectBC.x),alpha = 0.5, outlier.shape = 1)+
  scale_color_manual(values=patcol)+ggtitle("Viabilities with Ibrutinib")+coord_flip()

ggIdea<-ggplot(df )+
  geom_point(aes(x=reorder(BDrug,effectBC.y, median), y=effectBC.y, col=PatientID), alpha=0.7)+
  geom_boxplot(aes(x=reorder(BDrug,effectBC.y, median), y=effectBC.y),alpha = 0.5, outlier.shape = 1)+
  scale_color_manual(values=patcol)+ggtitle("Viabilities with Idelalisib")+coord_flip()

grid.arrange(ggIbru, ggIdea, ncol=2)
dev.off()

#correlation between two base drugs across all patients for comnination effect with Ibrutinib or Idelalisib
corIdea<-cor(matIdea)
corIbru<-cor(matIbru)
pdf(file.path(outputdir,paste("HeatmapDrugCor_IbruVsIdeal", feature, status,".pdf", sep="_")), width=15, height = 15)
pheatmap(corIdea, main="Correlation of viability values across patients for combination with CAL-101")
pheatmap(corIbru, main="Correlation of viability values across patients for combination with Ibrutinib100")
corCombiIdea<-corIdea
corCombiIdea[lower.tri(corCombiIdea, diag = F)]<-0

corCombiIbru<-corIbru
corCombiIbru[upper.tri(corCombiIbru, diag = F)]<-0
corCombi<-corCombiIdea+corCombiIbru
diag(corCombi)<-1

pheatmap(corCombi, cluster_rows = F, cluster_cols =F, main="Correlation of viability values across patients for Ibrutinib (lower triangle) and CAL-101 (upper triangle)")
dev.off()
}

```

```{r}
PlotHeatmapIbruIdea4Subgroup("IGHV",0)
PlotHeatmapIbruIdea4Subgroup("IGHV",1)
PlotHeatmapIbruIdea4Subgroup("TP53",0)
PlotHeatmapIbruIdea4Subgroup("TP53",1)
```


#PCA plots

```{r, eval=T}
pdf(file.path(outputdir,"PCAplots.pdf"))
for(drC in unique(df4anaAvreplicates$CDrugAbrv)){
df<-filter(df4anaAvreplicates, CDrugAbrv==drC)
df$BDrug<-paste(df$BDrugName, df$BDrugConcId, sep="_")

dfsel<-select(df, BDrug, effectBC, PatientID)
df4melt<-select(dfsel,PatientID,BDrug, effectBC )
mat<-melt(df4melt,id.vars=c("PatientID", "BDrug"), measure.vars = "effectBC") %>%
  acast(PatientID~BDrug)
#remove Na (patients being outlier in at least one screen)
noOutl<-apply(mat,1,function(r) !any(is.na(r))) 
mat<-mat[noOutl,]

title<-drC
if(any(mat>6)) {mat<-mat[-which(mat>6,arr.ind=T)[1],]; title<-paste(drC, "(One outlier removed)")}
pca.out<-prcomp(mat)
VarExpl<-round(pca.out$sdev/sum(pca.out$sdev),3)*100
df4PCA<-data.frame(PC1=pca.out$x[,1],PC2=pca.out$x[,2], IGHV=as.factor(IGHVData[rownames(pca.out$x),]), TP53=as.factor(genData[rownames(pca.out$x),"TP53"]))
gg<-ggplot(df4PCA, aes(x=PC1, y=PC2, col=IGHV, shape=TP53))+geom_point()+ggtitle(title)+coord_fixed()+xlab(paste(VarExpl[1], "% of variance explained"))+ylab(paste(VarExpl[2], "% of variance explained"))
print(gg)
}
dev.off()
```



#Curves of viabilities seperated for M-CLL and U-CLL
```{r}
load("df4CI.RData")

pdf(file.path(outputdir,"CurvesSingleCombiPerc_MCLL.pdf"), width=15, height=18)
df4CIMCLL<-filter(df4CI, PatientID %in% rownames(IGHVData[IGHVData==1 &!is.na(IGHVData),, drop=F]))
for(drC in unique(df4CI$CDrugAbrv)){
MedianDf4CI<-df4CIMCLL%>% filter(CDrugAbrv==drC) %>% 
  group_by( BDrugName, BDrugConcId) %>%
  dplyr::summarise(medianViabB=median(viabBper),medianViabC=median(viabCper),
                   medianViabBC=median(viabBCper),medianthViabBC=median(thviabBCper),
                   MADViabB=mad(viabBper),MADViabC=mad(viabCper),
                   MADViabBC=mad(viabBCper), MADthViabBC=mad(thviabBCper))
                                                                                            MedianDf4CI$BDrugConcId<-factor(MedianDf4CI$BDrugConcId, levels=rev(levels(MedianDf4CI$BDrugConcId)))          
gg<-ggplot(ungroup(MedianDf4CI), aes(x=BDrugConcId, group=BDrugName)) +
  geom_line(aes(y=medianViabBC, colour="BC"))+geom_point(aes(y=medianViabBC, colour="BC"))+geom_pointrange(aes(y=medianViabBC, ymin=medianViabBC-MADViabBC,ymax=medianViabBC+MADViabBC, colour="BC"))+
  geom_point(aes(y=medianViabB, colour="B"))+geom_line(aes(y=medianViabB, colour="B"))+geom_pointrange(aes(y=medianViabB, ymin=medianViabB-MADViabB,ymax=medianViabB+MADViabB, colour="B"))+
  geom_point(aes(y=medianViabC, colour="C"))+geom_line(aes(y=medianViabC, colour="C"))+geom_pointrange(aes(y=medianViabC, ymin=medianViabC-MADViabC,ymax=medianViabC+MADViabC, colour="C"))+
  geom_point(aes(y=medianthViabBC, colour="B*C"))+geom_line(aes(y=medianthViabBC, colour="B*C"))+geom_pointrange(aes(y=medianthViabBC,ymin=medianthViabBC-MADthViabBC,ymax=medianthViabBC+MADthViabBC, colour="B*C"))+
 ggtitle(drC)+facet_wrap(~BDrugName, ncol=5, scales="free_y")+xlab("concentration of base drug") +ylab("Median viability (% of control)")+
    theme_bw(base_size=18)#+scale_color_manual(values=brewer.pal(5,"Set1")[c(1:3,5)])
print(gg)
}
dev.off()

pdf(file.path(outputdir,"CurvesSingleCombiPerc_UCLL.pdf"), width=15, height=18)
df4CIUCLL<-filter(df4CI, PatientID %in% rownames(IGHVData[IGHVData==0 &!is.na(IGHVData),, drop=F]))
for(drC in unique(df4CI$CDrugAbrv)){
MedianDf4CI<-df4CIUCLL%>% filter(CDrugAbrv==drC) %>% 
  group_by( BDrugName, BDrugConcId) %>%
  dplyr::summarise(medianViabB=median(viabBper),medianViabC=median(viabCper),
                   medianViabBC=median(viabBCper),medianthViabBC=median(thviabBCper),
                   MADViabB=mad(viabBper),MADViabC=mad(viabCper),
                   MADViabBC=mad(viabBCper), MADthViabBC=mad(thviabBCper))
                                                                                            MedianDf4CI$BDrugConcId<-factor(MedianDf4CI$BDrugConcId, levels=rev(levels(MedianDf4CI$BDrugConcId)))          
gg<-ggplot(ungroup(MedianDf4CI), aes(x=BDrugConcId, group=BDrugName)) +
  geom_line(aes(y=medianViabBC, colour="BC"))+geom_point(aes(y=medianViabBC, colour="BC"))+geom_pointrange(aes(y=medianViabBC, ymin=medianViabBC-MADViabBC,ymax=medianViabBC+MADViabBC, colour="BC"))+
  geom_point(aes(y=medianViabB, colour="B"))+geom_line(aes(y=medianViabB, colour="B"))+geom_pointrange(aes(y=medianViabB, ymin=medianViabB-MADViabB,ymax=medianViabB+MADViabB, colour="B"))+
  geom_point(aes(y=medianViabC, colour="C"))+geom_line(aes(y=medianViabC, colour="C"))+geom_pointrange(aes(y=medianViabC, ymin=medianViabC-MADViabC,ymax=medianViabC+MADViabC, colour="C"))+
  geom_point(aes(y=medianthViabBC, colour="B*C"))+geom_line(aes(y=medianthViabBC, colour="B*C"))+geom_pointrange(aes(y=medianthViabBC,ymin=medianthViabBC-MADthViabBC,ymax=medianthViabBC+MADthViabBC, colour="B*C"))+
 ggtitle(drC)+facet_wrap(~BDrugName, ncol=5, scales="free_y")+xlab("concentration of base drug") +ylab("Median viability (% of control)")+
    theme_bw(base_size=18)#+scale_color_manual(values=brewer.pal(5,"Set1")[c(1:3,5)])
print(gg)
}
dev.off()
```


#SI heatmap durg by drug seperated between M-CLL and U-CLL
```{r, eval=F, echo=F}
DrugByDrugMatrixMean<-do.call(cbind, CombiBenefitMeanDiff)
DrugByDrugMatrixABC<-do.call(cbind, CombiBenefitABCMedian)
pheatmap(DrugByDrugMatrixABC)
#remove IPI145
# ipi145IDX<-which(rownames(DrugByDrugMatrixABC)=="IPI-145")
ipi145IDY<-which(colnames(DrugByDrugMatrixABC)=="IPI145")
DrugByDrugMatrixABC<-DrugByDrugMatrixABC[,-ipi145IDY]

#remove IPI145
# ipi145IDX<-which(rownames(DrugByDrugMatrixMean)=="IPI-145")
ipi145IDY<-which(colnames(DrugByDrugMatrixMean)=="IPI145")
DrugByDrugMatrixMean<-DrugByDrugMatrixMean[,-ipi145IDY]

pdf(file.path(outputdir, "HeatmapSynergyDrugByDrug.pdf"))
pheatmap(DrugByDrugMatrixABC, main = "Based on normalized area between \n viability curve of drug combination \n and theoretical additive viability curve \n (positive values=more than additive, \n negative values= less than additive)", clustering_distance_rows="correlation",
         clustering_distance_cols="correlation")
pheatmap(DrugByDrugMatrixMean, main = "Based on mean difference across conentrations \n between viability median across patients of drug combination \n and theoretical additive model \n (positive values=more than additive, \n negative values= less than additive)", clustering_distance_rows="correlation",
         clustering_distance_cols="correlation")
dev.off()
```

#ABC IGHV-specific
```{r, eval=T}
getDrugByDrugSI<-function(IGHV="all",  ExcludeIpi145=T){
dfAggr<-select(df4anaAvreplicates,CDrugAbrv, BDrugName, PatientID, effectC)
if(IGHV=="MCLL"){
dfAggr<- filter(dfAggr, PatientID%in%rownames(IGHVData[which(IGHVData==1),, drop=F]))
}else if(IGHV=="UCLL"){
  dfAggr<- filter(dfAggr, PatientID%in%rownames(IGHVData[which(IGHVData==0),, drop=F]))
}
dfAggr<-dfAggr[!duplicated(dfAggr),]
dfAggr$ABC<-sapply(1:nrow(dfAggr), function(i){
dfidx<-dfAggr[i,] 
subDF<-df4anaAvreplicates  %>%  
  filter(CDrugAbrv==as.character(dfidx$"CDrugAbrv"),
         BDrugName==as.character(dfidx$"BDrugName"),
         PatientID==as.character(dfidx$"PatientID"))
#claculate area between the curves and normalize by highest concentration used
#minus because concentrations ordered from high to low
(-trapz(subDF$BDrugConc, subDF$effectB*subDF$effectC)-(-trapz(subDF$BDrugConc, subDF$effectBC)))/as.numeric(filter(DrugBase,Substance==dfidx$"BDrugName")[4])
}) 
dfAggr<-cbind(dfAggr,t(sapply(1:nrow(dfAggr), function(i){
dfidx<-dfAggr[i,] 
subDF<-df4anaAvreplicates  %>%  
  filter(CDrugAbrv==as.character(dfidx$"CDrugAbrv"),
         BDrugName==as.character(dfidx$"BDrugName"),
         PatientID==as.character(dfidx$"PatientID"))
effectB<-subDF$effectB
names(effectB)<-paste("effectB_",subDF$BDrugConcId, sep="")
effectB
}) ))

medianABC<- dfAggr %>% group_by(CDrugAbrv, BDrugName) %>% dplyr::summarise(median(ABC))
if(ExcludeIpi145) medianABC<-filter(medianABC, CDrugAbrv!='IPI145' )

return(acast(medianABC, CDrugAbrv~BDrugName))
}

medianABC_total<-getDrugByDrugSI()
medianABC_UCLL<-getDrugByDrugSI(IGHV="UCLL")
medianABC_MCLL<-getDrugByDrugSI(IGHV="MCLL")

pdf(file.path(outputdir,"SIHeatmapDrugByDrug.pdf"))
pheatmap(t(medianABC_total), main = "Heatmap showing median Synergy Index (ABC) \n across all patients", clustering_distance_rows="correlation",
         clustering_distance_cols="correlation",
         breaks=seq(-0.25,0.3,0.005),
         color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(120))
pheatmap(t(medianABC_MCLL),  main = "Heatmap showing median Synergy Index (ABC) \n across M-CLL patients", clustering_distance_rows="correlation",
         clustering_distance_cols="correlation",
         breaks=seq(-0.25,0.3,0.005),
           color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(120))
pheatmap(t(medianABC_UCLL), main = "Heatmap showing median Synergy Index (ABC) \n across U-CLL patients", clustering_distance_rows="correlation",
         clustering_distance_cols="correlation",
         breaks=seq(-0.25,0.3,0.01),
         color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(60))

dev.off()
```




# Patient-patients correlation matrix for combinaiton with Ibrutinib
Not saling for each drugs -> more variable drugs contribtue more (ok as they are measured on the same scale?)

```{r}
dfIbru<- df4anaAvreplicates %>% filter(CDrugAbrv=="Ibrutinib100") %>% mutate(drB=paste(BDrugName,BDrugConcId, sep="_"))
moltendfIbru<-melt(dfIbru,id.vars=c("PatientID", "drB"), measure.vars = "effectBC") 
PDmatIbru<-acast(moltendfIbru, PatientID~drB)

#remove Na (patients being outlier in at least one screen)
noOutl<-apply(PDmatIbru,1,function(r) !any(is.na(r)))
PDmatIbru<-PDmatIbru[noOutl,]

IGHVannot<-IGHVData[rownames(PDmatIbru),, drop=F]

corPP<-cor(t(PDmatIbru))
callbackPC = function(hc, mat){
    sv = svd(t(mat))$v[,1]
    dend = reorder(as.dendrogram(hc), wts = sv)
    as.hclust(dend)
}

ghm<-pheatmap(corPP,
              main="Patient-patient similarity for combination with Ibrutinib \n (using all base drugs and all concentrations)",
              clustering_distance_rows = "correlation",
              clustering_distance_cols = "correlation" )

#annotate with mutation background
patsInOrder<-ghm$tree_row$labels[ghm$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnno<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnno<-ggplotGrob(ggMutAnno)

heights4Fig<-c(2,1,10,3)
widths4Fig<-c(1,10,1,1,2,1)
gt<-gtable(width<-unit(widths4Fig, "in"), heights = unit(heights4Fig,"in"))
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[1]], 1, 2) #main
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[2]], 2, 2) #dendrogram columns
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[3]], 3, 1) #dendrogram rows
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[4]], 3, 2) #matrix
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[5]], 4, 2) #column names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[6]], 3, 3) #row names
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[4]], 3, 5) #mutation matrix
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[5]], 4, 5) #mutation names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[7]], 3, 4, clip="off") #legend
pdf(file.path(outputdir,"PatbyPatIbru.pdf"), width=sum(widths4Fig), height=sum(heights4Fig))
grid.draw(gt)
dev.off()
```

Only Cluster for a subset of base compounds (like Gosia did in 2015.05.13) (8 drugs: idelalisib, duvelisib, everolimus, selumetinib, afatinib, AZD7762, PF 477736, navitoclax)
Gosia: idelalisib - c5, duvelisib - c4, everolimus - c3, selumetinib - c3, afatinib - c2, AZD7762 - c1, PF 477736 - c2, navitoclax - c3 - here all concentrations used
```{r}
dfIbruSubset<- df4anaAvreplicates %>% filter(CDrugAbrv=="Ibrutinib100" & BDrugID %in%c("D003","D082","D063","D012","D013","D020","D078","D001"))%>% mutate(drB=paste(BDrugName,BDrugConcId, sep="_"), drBid=paste(BDrugID, BDrugConcId, sep="_")) #%>% filter(drBid %in%c("D003_c5","D082_c4","D063_c3","D012_c3","D013_c2","D020_c1","D078_c2","D001_c3"))
moltendfIbru<-melt(dfIbruSubset,id.vars=c("PatientID", "drB"), measure.vars = "effectBC") 
PDmatIbru<-acast(moltendfIbru, PatientID~drB)
PDmatIbru<-PDmatIbru[rownames(PDmatIbru)!="P0462",] #outlier in Navitoclax_c3

#remove Na (patients being outlier in at least one screen)
noOutl<-apply(PDmatIbru,1,function(r) !any(is.na(r)))
PDmatIbru<-PDmatIbru[noOutl,]

IGHVannot<-IGHVData[rownames(PDmatIbru),, drop=F]

corPP<-cor(t(PDmatIbru))
callbackPC = function(hc, mat){
    sv = svd(t(mat))$v[,1]
    dend = reorder(as.dendrogram(hc), wts = sv)
    as.hclust(dend)
}

ghm<-pheatmap(corPP,
              main="Patient-patient similarity for combination with Ibrutinib \n (using a subset of 8 drugs: idelalisib, duvelisib, everolimus, selumetinib, afatinib, AZD7762, PF 477736, navitoclax)",
              clustering_distance_rows = "correlation",
              clustering_distance_cols = "correlation" )

#annotate with mutation background
patsInOrder<-ghm$tree_row$labels[ghm$tree_row$order]
mut<-cbind(IGHV=IGHVData[patsInOrder,],
           genData[patsInOrder,c("TP53","trisomy12","del13q14","del11q22.3","SF3B1")])
mutDF<-MolMatrix2DF(mut)
ggMutAnno<-ggplot(data=mutDF, aes(y=factor(Patient, level=rev(patsInOrder)), x=Feature, fill=Value)) + geom_tile() + scale_fill_manual(values=c("0"="white","1"="black", "NA"="grey90"), name="mutations", labels=c("wt","mut","NA")) + ylab("") + xlab("") + geom_vline(xintercept=seq(0.5,length(unique(mutDF$Feature))+1,1), colour="grey60") + geom_hline(yintercept=seq(0.5,length(unique(mutDF$Patient))+1,1), colour="grey60") + theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5), axis.text = element_text(colour="black")) + scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) + theme(axis.ticks=element_blank(), axis.text=element_text(margin=unit(0.5,"cm")), legend.key = element_rect(colour = "black"), legend.text=element_text(size=16), legend.title=element_text(size=16))
grobMutAnno<-ggplotGrob(ggMutAnno)

heights4Fig<-c(2,1,10,3)
widths4Fig<-c(1,10,1,1,2,1)
gt<-gtable(width<-unit(widths4Fig, "in"), heights = unit(heights4Fig,"in"))
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[1]], 1, 2) #main
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[2]], 2, 2) #dendrogram columns
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[3]], 3, 1) #dendrogram rows
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[4]], 3, 2) #matrix
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[5]], 4, 2) #column names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[6]], 3, 3) #row names
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[4]], 3, 5) #mutation matrix
gt<-gtable_add_grob(gt, grobMutAnno$grobs[[5]], 4, 5) #mutation names
gt<-gtable_add_grob(gt, ghm$gtable$grobs[[7]], 3, 4, clip="off") #legend
pdf(file.path(outputdir,paste("PatbyPatIbruSubsetDrugs.pdf", sep="")), width=sum(widths4Fig), height=sum(heights4Fig))
grid.draw(gt)
dev.off()

```



#Comparions BCR inhibitors seperated fo U-CLL and M-CLL
Scatterplots comparing the effect of 2 BCR inhibitors across all base drugs
```{r}
PlotComparsionCombiCDrugs<-function(CDrugAbrv.x, CDrugAbrv.y, range = c(0,1.25), IGHVst=0){
df.x<-filter(df4anaAvreplicates, CDrugAbrv==CDrugAbrv.x& PatientID%in%names(IGHVData[which(IGHVData==IGHVst),]))
df.y<-filter(df4anaAvreplicates, CDrugAbrv==CDrugAbrv.y& PatientID%in%names(IGHVData[which(IGHVData==IGHVst),]))
df<-merge(df.x, df.y, by=c("PatientID","BDrugID", "BDrugConcId",
                                               "BDrugConc", "BDrugName"))
effect.x<-df$effectBC.x
effect.y<-df$effectBC.y

filename<-paste("ComparisonCombi", CDrugAbrv.x, CDrugAbrv.y,"IGHV=",IGHVst,".pdf", sep="_")
  pdf(file.path(outputdir, filename), width=10, height=12)
  df$cens.x<-ifelse(effect.x<range[1], range[1], ifelse(effect.x>range[2], range[2], effect.x))
  df$cens.y<-ifelse(effect.y<range[1], range[1], ifelse(effect.y>range[2], range[2], effect.y))
  df$shape<-ifelse(df$cens.x!=effect.x | df$cens.y!=effect.y, "censored", "uncensored")

  gg1<-ggplot(df, aes(x=cens.x, y=cens.y, col=BDrugName, shape=shape))+geom_point()+xlab(paste("Combination effect with", CDrugAbrv.x, "\n (viability relative to DMSO control)"))+ylab(paste("Combination effect with", CDrugAbrv.y, "\n (viability relative to DMSO control)"))+    scale_shape_manual(values=c(censored=2, uncensored=19)) + annotate("text", x=range[1]+0.1, y=range[2]-0.1, label=paste("rho^2 ==", (round(cor(df$effectBC.x,df$effectBC.y)^2,2))), parse=T)

  print(gg1)
  
  gg2<-ggplot(df, aes(x=cens.x, y=cens.y, col=BDrugName, shape=shape))+geom_point()+
    xlab("Combination effect with Ibrutinib100 \n (viability relative to DMSO control)")+
    ylab("Combination effect with CAL-101 \n (viability relative to DMSO control)")+
    facet_wrap(~BDrugName)+    scale_shape_manual(values=c(censored=2, uncensored=19))+
    annotate("text", x=range[1]+0.3, y=range[2]-0.1, label=paste("rho^2 ==", (round(dplyr::summarize(group_by(df, BDrugName), cor=cor(effectBC.x, effectBC.y))$cor,2))), parse=T, size=3) +guides(col=guide_legend(ncol=1))
  print(gg2)
  df_remOutlier<-df

  df_directcomp<-data.frame(effect=c(df_remOutlier$effectBC.x,
                                     df_remOutlier$effectBC.y),
                            drug=c(as.character(df_remOutlier$CDrugAbrv.x),
                                   as.character(df_remOutlier$CDrugAbrv.y)))
  median<-df_directcomp%>%group_by(drug)%>% dplyr::summarise(median(effect))
  t.out<-t.test(df_directcomp$effect ~ df_directcomp$drug, var.equal=TRUE)
  
gg3<-ggplot(df_directcomp, aes(x=drug, y=effect))+geom_point(position="jitter", alpha=0.4, col="forestgreen")+geom_boxplot(alpha=0.4) + annotate("text", label=paste("p=",round(t.out$p.value,3)), x = 1.5, y = 2, size = 4)+ annotate("text", label=paste("median=",as.numeric(median[1,2])), x = 1, y = 1.5, size = 4)+ annotate("text", label=paste("median=", as.numeric(median[2,2])), x = 2, y = 1.5, size = 4)
print(gg3)

dev.off()
}
  
```

```{r}
PlotComparsionCombiCDrugs(CDrugAbrv.x="Ibrutinib100", CDrugAbrv.y = "CAL101", range = c(0,1.2), IGHVst = 0)
PlotComparsionCombiCDrugs(CDrugAbrv.x="Ibrutinib100", CDrugAbrv.y = "CAL101", range = c(0,1.2), IGHVst = 1)
```
